src/main/java/bruno/TaskList.java

```diff
@@ -0,0 +1,255 @@
+package bruno;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+import bruno.exceptions.BrunoEmptyException;
+import bruno.exceptions.BrunoException;
+import bruno.exceptions.BrunoIndexOutOfBoundsException;
+import bruno.exceptions.BrunoIntegerMismatchException;
+import bruno.exceptions.BrunoMissingDeadlineException;
+import bruno.exceptions.BrunoMissingEventException;
+import bruno.exceptions.BrunoNegativeArgException;
+import bruno.task.Deadline;
+import bruno.task.Event;
+import bruno.task.Task;
+import bruno.task.ToDo;
+
+/**
+ * The TaskList class handles all functionality relating to the tasks of tasks, such as addition, deletion,
+ * update and display of tasks.
+ */
+public class TaskList {
+
+    private List<Task> tasks;
+    private Storage storage;
+    private UI ui;
+
+    TaskList(Storage storage, UI ui) {
+        tasks = new ArrayList<>();
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    /**
+     * Adds a task of type ToDo to the tasks of tasks.
+     *
+     * @param task The user input for task.
+     * @throws BrunoException Thrown if user does not provide description for task.
+     */
+    public String addToDo(String task) throws BrunoException {
+        if (task.split(" ").length == 1) {
+            throw new BrunoEmptyException(task.split(" ")[0]);
+        }
+        tasks.add(new ToDo(task.substring(task.indexOf(" ") + 1)));
+        String taskInfo = "Woof. I have added this task:\n" + tasks.get(tasks.size() - 1).getString();
+        storage.writeToFile(this);
+        return taskInfo;
+    }
+
+    /**
+     * Adds a task of type Deadline to the tasks of tasks.
+     *
+     * @param task The user input for the task.
+     * @throws BrunoException Thrown if user does not provide description or deadline date/time.
+     */
+    public String addDeadline(String task) throws BrunoException {
+        if (task.split(" ").length == 1 || task.indexOf('/') == 9) {
+            throw new BrunoEmptyException(task.split(" ")[0]);
+        }
+        if (!task.substring(task.indexOf("deadline") + 1).contains("/by")) {
+            throw new BrunoMissingDeadlineException();
+        }
+        tasks.add(new Deadline(task.substring(task.indexOf(' ') + 1, task.indexOf('/') - 1),
```

Comment:
for all the `addTask()` methods in this class, maybe can consider extracting out the parsing methods like `substring()` and `indexOf()` to make the code more readable?

```suggestion
        String description = task.substring(task.indexOf(' ') + 1, task.indexOf('/') - 1);
        String by = task.substring(task.lastIndexOf('/') + 4);
        tasks.add(new Deadline(description, by));
```